redhat-developer/odo
# Refactor component commands to follow complete-validate-run pattern

Refactor component commands to follow complete-validate-run pattern

This PR is an internal code refactoring change that makes component
commands follow complete-validate-run pattern.

Commit 1: Refactor watch command
Commit 2: Refactor create command
Commit 3: Refactor log and list commands

To test changes in this PR,
1. Run all component commands like
    a. create
    b. push
    c. watch
    d. log
    e. list
    ...
    along with their respective auto-completion
2. Check travis CI runs

fixes #1190
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>
